include/adaflo/sharp_interface_util.h

```diff
@@ -0,0 +1,896 @@
+              {
+                const auto point = fe_eval.quadrature_point(q);
+                points.emplace_back(point[0], point[1]);
+              }
+          }
+      }
+
+      points.push_back(points.front());
+
+      boost::geometry::assign_points(poly, points);
+    }
+
+
+    template <int dim>
+    double
+    within(
+      const boost::geometry::model::polygon<boost::geometry::model::d2::point_xy<double>>
+        &               polygon,
+      const Point<dim> &point)
+    {
+      boost::geometry::model::d2::point_xy<double> p(point[0], point[1]);
+      return boost::geometry::within(p, polygon);
+    }
+    template <int dim>
+    VectorizedArray<double>
+    within(
+      const boost::geometry::model::polygon<boost::geometry::model::d2::point_xy<double>>
+        &                                        polygon,
+      const Point<dim, VectorizedArray<double>> &points)
+    {
+      VectorizedArray<double> result;
+
+      for (unsigned int v = 0; v < VectorizedArray<double>::size(); ++v)
+        {
+          boost::geometry::model::d2::point_xy<double> p(points[0][v], points[1][v]);
+          result[v] = boost::geometry::within(p, polygon);
+        }
+
+      return result;
+    }
+
+    template <int dim, int spacedim, typename VectorType>
+    void
+    within(const MappingFEField<dim, spacedim> &mapping,
+           const DoFHandler<dim, spacedim> &    dof_handler,
+           const Mapping<spacedim> &            background_mapping,
+           const DoFHandler<spacedim> &         background_dof_handler,
+           VectorType &                         force_vector_sharp_interface)
+    {
+      typedef boost::geometry::model::d2::point_xy<double> point_type;
+      typedef boost::geometry::model::polygon<
+        boost::geometry::model::d2::point_xy<double>>
+        polygon_type;
+
+      polygon_type poly;
+
+      construct_polygon(mapping, dof_handler, poly);
+
+      {
+        FEValues<spacedim> fe_eval(
+          background_mapping,
+          background_dof_handler.get_fe(),
+          background_dof_handler.get_fe().get_unit_support_points(),
+          update_quadrature_points);
+
+        for (const auto &cell : background_dof_handler.active_cell_iterators())
+          {
+            fe_eval.reinit(cell);
+
+            Vector<double> vec(fe_eval.n_quadrature_points);
+
+            for (const auto q : fe_eval.quadrature_point_indices())
+              {
+                vec[q] = static_cast<double>(within(poly, fe_eval.quadrature_point(q)));
+              }
+
+            cell->set_dof_values(vec, force_vector_sharp_interface);
+          }
+      }
+    }
+  } // namespace GridTools
+
+  namespace GridGenerator
+  {
+    template <int dim>
+    class MarchingCubeAlgorithm
+    {
+    public:
+      MarchingCubeAlgorithm(const Mapping<dim, dim> &      mapping,
+                            const FiniteElement<dim, dim> &fe,
+                            const unsigned int             n_subdivisions)
```

Comment:
@mschreter I will be adding new features to this PR in the next hours. So don't wonder that GitHub might complain that the code is outdated. But there (probably) won't be any changes in `FrontTrackingSolver` and `MixedLevelSetSolver`.